internal/models/argo.go

```diff
@@ -168,6 +187,88 @@ func (app *Application) ListUnhealthyResources() []string {
 	return list
 }
 
+// IsManagedByWatcher checks if the application is managed by the watcher.
+// It checks if the application's metadata contains the "argo-watcher/managed" annotation with the value "true".
+func (app *Application) IsManagedByWatcher() bool {
+	if app.Metadata.Annotations == nil {
+		return false
+	}
+	return app.Metadata.Annotations[managedAnnotation] == "true"
+}
+
+func (app *Application) processAppAnnotations(annotations map[string]string, task *Task) *updater.ArgoOverrideFile {
+	overrideFileContent := updater.ArgoOverrideFile{}
+	managedImages := extractManagedImages(annotations)
+
+	if len(managedImages) == 0 {
+		log.Error().Msgf("%s annotation not found, skipping image update", managedImagesAnnotation)
+		return nil
+	}
+
+	for _, image := range task.Images {
+		for appAlias, appImage := range managedImages {
+			if image.Image == appImage {
+				if tagPath, exists := annotations[fmt.Sprintf(managedImageTagPattern, appAlias)]; exists {
+					overrideFileContent.Helm.Parameters = append(overrideFileContent.Helm.Parameters, updater.ArgoParameterOverride{
+						Name:        tagPath,
+						Value:       image.Tag,
+						ForceString: true,
+					})
+				} else {
+					log.Error().Msgf("%s annotation not found, skipping image %s update", fmt.Sprintf(managedImageTagPattern, appAlias), image.Image)
+				}
+			}
+		}
+	}
+
+	return &overrideFileContent
+}
+
+func (app *Application) UpdateGitImageTag(task *Task) {
+	if app.Spec.Source.Path == "" {
+		log.Warn().Str("id", task.Id).Msgf("No path found for app %s, unsupported Application configuration", app.Metadata.Name)
+		return
+	}
+
+	releaseOverrides := app.processAppAnnotations(app.Metadata.Annotations, task)
+	if releaseOverrides == nil {
+		log.Warn().Str("id", task.Id).Msgf("No release overrides found for app %s", app.Metadata.Name)
+		return
+	}
+
+	git := updater.GitRepo{
+		RepoURL:    app.Spec.Source.RepoURL,
+		BranchName: app.Spec.Source.TargetRevision,
+		Path:       app.Spec.Source.Path,
+	}
+
+	if err := git.Clone(); err != nil {
+		log.Error().Str("id", task.Id).Msgf("Failed to clone git repository %s", app.Spec.Source.RepoURL)
+		return
+	}
+
+	if err := git.UpdateApp(app.Metadata.Name, releaseOverrides); err != nil {
+		log.Error().Str("id", task.Id).Msgf("Failed to update git repository %s", app.Spec.Source.RepoURL)
+		return
+	}
+}
+
+// extractManagedImages extracts the managed images from the application's annotations.
+// It returns a map of the managed images, where the key is the application alias and the value is the image name.
+func extractManagedImages(annotations map[string]string) map[string]string {
+	managedImages := map[string]string{}
+
+	for annotation, value := range annotations {
+		if annotation == managedImagesAnnotation {
+			for _, image := range strings.Split(value, ",") {
+				managedImages[strings.Split(image, "=")[0]] = strings.Split(image, "=")[1]
```

Comment:
We need basic validation here; in case the format of the image string is wrong.
Same as with previous comments, errors should be propagated and should fail the deployment